### ISSUE TITLE
Improve Sourcegraph References section with reference preview before navigation

### ISSUE THREAD

**Issue by cosmonot1** [sourcegraph/sourcegraph#2343]
When viewing the list of references, a ~3 line preview is provided. There are cases where the code formatting around the 3 lines shown causes the snippet to not contain much useful information about the reference to the function call/variable. Image shown below.

![image](https://user-images.githubusercontent.com/7645379/52977283-8d8aea80-3381-11e9-8d0f-482a50a8f569.png)

You can see from this image, the only details we get from the References tab is the line number and the call to the function. The surrounding code is either whitespace or closing braces/parentheses. The only way to get more information from the snippet is by clicking the snippet and opening the file/section of code the snippet references. If you click on a reference, realize it was the wrong one, and return to the references section, Sourcegraph takes time to reload the references list. This is a little tedious and required lots of user interaction waiting for view changes. I think an "expand reference" feature would be very helpful. This feature, on button click or reference list item hover, would open a modal or expand the snippet in the references list to show the ~10-20 lines of code that the reference links to. This preview would help users make sure that they follow the correct reference without forcing them to click on it.

**Comment by sqs** [comment]
@cosmonot1 Thanks for posting this!

Do you like the VS Code references UI? When you click a reference location in the right-hand column, it shows a peek view of the entire file (centered at the reference location you clicked).

![image](https://user-images.githubusercontent.com/1976/52991172-66074280-33c0-11e9-9125-9ae7daa238ad.png)

We will be rethinking the references  UI soon to accommodate other new requirements (such as #1174), and we'll take this into consideration then.

**Comment by felixfbecker** [comment]
@cosmonot1 which language were you looking at that was slow to load references?

**Comment by cosmonot1** [comment]
I was looking at JavaScript.

**Comment by cosmonot1** [comment]
Yes I like the VS Code UI. Having a quick snippet pop up like that would be
pretty helpful IMO.

**Comment by aidaeology** [comment]
This applies to all code snippets not just references.
